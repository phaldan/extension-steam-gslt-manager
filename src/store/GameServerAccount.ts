import { action, observable } from 'mobx';

export default class GameServerAccount {
  public steamid: string;
  public appid: number;

  @observable
  public expired: boolean = false;

  @observable
  public lastLogon?: Date;

  @observable
  public memo: string;

  @observable
  public token: string;

  @observable
  public locked: boolean = false;

  constructor({ steamid, appid, token, memo, lastLogon, expired }) {
    this.steamid = steamid;
    this.appid = appid;
    this.token = token;
    this.memo = memo;
    this.lastLogon = lastLogon ? new Date(lastLogon) : undefined;
    this.expired = expired;
  }

  @action
  public toggleLock(locked: boolean) {
    this.locked = locked;
  }

  @action
  public updateFromJson({ appid, token, memo, lastLogon, expired }) {
    this.appid = appid;
    this.token = token;
    this.memo = memo;
    this.lastLogon = lastLogon;
    this.expired = expired;
  }
}
