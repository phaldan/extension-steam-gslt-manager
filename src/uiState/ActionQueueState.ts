import { action, computed, observable } from 'mobx';
import ActionQueue from './ActionQueue';

export default class ActionQueueState {
  private removeDelay = 3000;

  @observable
  private queues: ActionQueue[] = [];

  @action
  public create(description: string, data: string[]): ActionQueue {
    const entry = new ActionQueue(this, this.removeDelay, description, data);
    this.queues.push(entry);
    return entry;
  }

  @computed
  public get running(): ActionQueue[] {
    return this.queues;
  }

  @action
  public async remove(id: string) {
    this.queues = this.queues.filter((q) => q.id !== id);
  }

  public setRemoveDelay(removeDelay: number): void {
    this.removeDelay = removeDelay;
  }
}
