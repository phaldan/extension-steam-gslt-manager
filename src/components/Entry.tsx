import copy from 'copy-to-clipboard';
import { observer } from 'mobx-react';
import * as React from 'react';
import { Button, Checkbox, FormControl, Glyphicon } from 'react-bootstrap';
import { FormattedDate } from 'react-intl';
import GameServerAccount from '../store/GameServerAccount';
import GsltStore from '../store/GsltStore';
import ActionQueueState from '../uiState/ActionQueueState';
import Edit from './Edit';

interface EntryProps {
  account: GameServerAccount;
  actions: ActionQueueState;
  enableRegenerate?: boolean;
  onToggle?: (selected: boolean, account: GameServerAccount) => void;
  selected?: boolean;
  store: GsltStore;
}

interface EntryState {
  enableDelete: boolean;
  enableRegenerate: boolean;
}

@observer
export default class Entry extends React.Component<EntryProps, EntryState> {
  public static defaultProps: Partial<EntryProps> = {
    selected: false,
  };

  constructor(props) {
    super(props);
    this.state = { enableDelete: true, enableRegenerate: true };
  }

  public render() {
    const { account, store, selected, actions } = this.props;
    const { token, appid, lastLogon, memo } = account;
    return (
      <tr className="align-middle">
        <td>
          <div className="form-inline entry-select">
            <Checkbox
              checked={this.props.selected}
              onChange={this.fireToggle}
              className="js-select"
            />
          </div>
        </td>
        <td>
          <FormControl type="text" value={token} readOnly={true} bsSize="small" />
        </td>
        <td className="js-appid">{appid}</td>
        <td className="text-center">
          {lastLogon ? this.formatDate(lastLogon) : <span className="text-danger">Never</span>}
        </td>
        <td className="js-memo">{memo}</td>
        <td className="text-right">
          <Button
            bsStyle="danger"
            bsSize="small"
            title="Remove"
            onClick={this.handleRemove}
            disabled={this.isRemoveDisabled}
            className="js-remove"
          >
            <Glyphicon glyph="trash" />
          </Button>
          <Button
            bsStyle="success"
            bsSize="small"
            title="Regenerate Token"
            onClick={this.handleRegenerate}
            disabled={this.isRegenerateDisabled}
            className="js-regenerate"
          >
            <Glyphicon glyph="refresh" />
          </Button>
          <Edit store={store} account={account} actions={actions} disabled={this.isRemoveDisabled}>
            <Glyphicon glyph="edit" />
          </Edit>
          <Button
            bsStyle="primary"
            bsSize="small"
            title="Copy to clipboard"
            onClick={this.copy}
            className="js-clipboard"
          >
            <Glyphicon glyph="duplicate" />
          </Button>
        </td>
      </tr>
    );
  }

  private fireToggle = (event) => {
    const { onToggle, account } = this.props;
    if (onToggle) {
      onToggle(event.target.checked, account);
    }
  }

  private formatDate(lastLogon) {
    return (
      <FormattedDate value={lastLogon} year="numeric" month="long" day="2-digit" />
    );
  }

  private handleRemove = async () => {
    const { account, store, actions } = this.props;
    if (!this.isRemoveDisabled) {
      this.setState({ enableDelete: false });
      const queue = actions.create('Deleting GSLT:', [account.steamid]);
      await store.removeAccount(account);
      queue.progress(account.steamid);
    }
  }

  private get isRemoveDisabled() {
    const { enableDelete } = this.state;
    const { account } = this.props;
    return !enableDelete || account.locked;
  }

  private handleRegenerate = async () => {
    const { account, store, actions } = this.props;
    if (!this.isRegenerateDisabled) {
      this.setState({ enableRegenerate: false });
      const queue = actions.create('Regenerate token for GSLT:', [account.steamid]);
      await store.regenerateToken(account);
      queue.progress(account.steamid);
      this.setState({ enableRegenerate: true });
    }
  }

  private get isRegenerateDisabled() {
    const { enableDelete, enableRegenerate } = this.state;
    return !enableRegenerate || this.isRemoveDisabled || !this.props.enableRegenerate;
  }

  private copy = () => {
    copy(this.props.account.token);
  }
}
