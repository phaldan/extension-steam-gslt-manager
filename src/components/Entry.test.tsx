import { mount, shallow } from 'enzyme';
import * as React from 'react';
import GameServerAccount from '../store/GameServerAccount';
import GsltStore from '../store/GsltStore';
import ActionQueueState from '../uiState/ActionQueueState';
import delay from '../utils/delay';
import Entry from './Entry';

jest.mock('copy-to-clipboard');

function createComponent(state, account, selected, regenerate, queue, callback = jest.fn()) {
  const element = (
    <Entry
      store={state}
      account={account}
      selected={selected}
      enableRegenerate={regenerate}
      onToggle={callback}
      actions={queue}
    />
  );
  return shallow(element);
}

class GsltStoreStub implements GsltStore {
  public removeAccountReturn: Promise<GameServerAccount>;
  public removeAccountMock = jest.fn();
  public regenerateTokenReturn: Promise<GameServerAccount>;
  public regenerateTokenMock = jest.fn();

  loadAccounts(): Promise<void> {
    throw new Error("Method not implemented.");
  }
  get tokenAccounts(): GameServerAccount[] {
    throw new Error("Method not implemented.");
  }
  removeAccount(account: GameServerAccount): Promise<GameServerAccount> {
    this.removeAccountMock(account);
    return this.removeAccountReturn;
  }
  removeAccounts(accounts: GameServerAccount[]): Promise<GameServerAccount>[] {
    throw new Error("Method not implemented.");
  }
  regenerateToken(account: GameServerAccount): Promise<GameServerAccount> {
    this.regenerateTokenMock(account);
    return this.regenerateTokenReturn;
  }
  regenerateTokens(accounts: GameServerAccount[]): Promise<GameServerAccount>[] {
    throw new Error("Method not implemented.");
  }
  updateMemo(account: GameServerAccount, memo: string): Promise<GameServerAccount> {
    throw new Error("Method not implemented.");
  }
  createAccounts(amount: number, appid: string, memo: string): Promise<void>[] {
    throw new Error("Method not implemented.");
  }
  get isLoggedIn(): boolean {
    throw new Error("Method not implemented.");
  }
  get isInitialized(): boolean {
    throw new Error("Method not implemented.");
  }
}

describe('<Entry>', () => {
  let store: GsltStoreStub;
  let queue: ActionQueueState;
  const copyToClipboard = require('copy-to-clipboard');

  function expectDisabledButtons(target, remove, regenerate, edit) {
    expect(target.find('.js-remove').props().disabled).toBe(remove);
    expect(target.find('.js-regenerate').props().disabled).toBe(regenerate);
    expect(target.find('Edit').props().disabled).toBe(edit);
  }

  function expectAccountContent(target, token, appid, hasLastLogon, memo) {
    expect(target.find('FormControl').props().value).toBe(token);
    expect(target.find('.js-appid').text()).toBe(appid);
    expect(target.find('FormattedDate').exists()).toBe(hasLastLogon);
    expect(target.find('.js-memo').text()).toBe(memo);
  }

  function expectAccountWithLastLogon(target, token, appid, lastLogon, memo) {
    expectAccountContent(target, token, appid, true, memo);
    expect(target.find('FormattedDate').props().value).toEqual(new Date(lastLogon));
  }

  function expectEdit(target, account) {
    expect(target.find('Edit').props().store).toBe(store);
    expect(target.find('Edit').props().account).toBe(account);
    expect(target.find('Edit').props().actions).toBe(queue);
  }

  function expectSnapshot(target) {
    expect(target.find('Checkbox')).toMatchSnapshot();
    expect(target.find('FormControl')).toMatchSnapshot();
    expect(target.find('FormattedDate')).toMatchSnapshot();
    expect(target.find('Button')).toMatchSnapshot();
  }

  beforeEach(() => {
    copyToClipboard.mockReset();
    store = new GsltStoreStub();
    queue = new ActionQueueState();
    queue.setRemoveDelay(0);
  });

  describe('selected and enabled regenerate', () => {
    let account: GameServerAccount;

    beforeEach(() => {
      account = new GameServerAccount({
        appid: 740,
        expired: false,
        lastLogon: '1995-12-17T03:24:00',
        memo: 'CSGO',
        steamid: '',
        token: '7FJS3VY2273L',
      });
    });

    test('initial state', () =>  {
      const target = createComponent(store, account, true, true, queue);

      expectDisabledButtons(target, false, false, false);
      expectAccountWithLastLogon(target, '7FJS3VY2273L', '740', '1995-12-17T03:24:00', 'CSGO');
      expect(target.find('.js-select').props().checked).toEqual(true);
      expectEdit(target, account);
      expect(queue.running).toHaveLength(0);
      expect(store.removeAccountMock).not.toBeCalled();
      expect(store.regenerateTokenMock).not.toBeCalled();
      expect(copyToClipboard).not.toBeCalled();
      expectSnapshot(target);
    });

    test('unselect', () =>  {
      const callback = jest.fn();
      const target = createComponent(store, account, true, true, queue, callback);
      target.find('.js-select').simulate('change', {target: {checked: false}});
      
      expectDisabledButtons(target, false, false, false);
      expectAccountWithLastLogon(target, '7FJS3VY2273L', '740', '1995-12-17T03:24:00', 'CSGO');
      expect(target.find('.js-select').props().checked).toEqual(true);
      expectEdit(target, account);
      expect(queue.running).toHaveLength(0);
      expect(callback).toBeCalled();
      expect(callback).toHaveBeenCalledWith(false, account);
      expect(store.removeAccountMock).not.toBeCalled();
      expect(store.regenerateTokenMock).not.toBeCalled();
      expect(copyToClipboard).not.toBeCalled();
      expectSnapshot(target);
    });

    test('click remove', async () =>  {
      store.removeAccountReturn = Promise.resolve(account);
      const target = createComponent(store, account, true, true, queue);
      target.find('.js-remove').simulate('click');

      await delay(10);
      expectDisabledButtons(target, true, true, true);
      expectAccountWithLastLogon(target, '7FJS3VY2273L', '740', '1995-12-17T03:24:00', 'CSGO');
      expect(target.find('.js-select').props().checked).toEqual(true);
      expectEdit(target, account);
      expect(queue.running).toHaveLength(0);
      expect(store.removeAccountMock).toBeCalledWith(account);
      expect(store.regenerateTokenMock).not.toBeCalled();
      expect(copyToClipboard).not.toBeCalled();
      expectSnapshot(target);
    });

    test('click remove with locked user', () =>  {
      account.toggleLock(true);
      const target = createComponent(store, account, true, true, queue);
      target.find('.js-remove').simulate('click');

      expectDisabledButtons(target, true, true, true);
      expectAccountWithLastLogon(target, '7FJS3VY2273L', '740', '1995-12-17T03:24:00', 'CSGO');
      expect(target.find('.js-select').props().checked).toEqual(true);
      expectEdit(target, account);
      expect(queue.running).toHaveLength(0);
      expect(store.removeAccountMock).not.toBeCalled();
      expect(store.regenerateTokenMock).not.toBeCalled();
      expect(copyToClipboard).not.toBeCalled();
      expectSnapshot(target);
    });

    test('click regenerate', async () =>  {
      store.regenerateTokenReturn = Promise.resolve(account);
      const target = createComponent(store, account, true, true, queue);
      target.find('.js-regenerate').simulate('click');

      await delay(10);
      expectDisabledButtons(target, false, false, false);
      expectAccountWithLastLogon(target, '7FJS3VY2273L', '740', '1995-12-17T03:24:00', 'CSGO');
      expect(target.find('.js-select').props().checked).toEqual(true);
      expectEdit(target, account);
      expect(queue.running).toHaveLength(0);
      expect(store.removeAccountMock).not.toBeCalled();
      expect(store.regenerateTokenMock).toBeCalledWith(account);
      expect(copyToClipboard).not.toBeCalled();
      expectSnapshot(target);
    });

    test('click copy to clipboard', () =>  {
      const target = createComponent(store, account, true, true, queue);
      target.find('.js-clipboard').simulate('click');

      expectDisabledButtons(target, false, false, false);
      expectAccountWithLastLogon(target, '7FJS3VY2273L', '740', '1995-12-17T03:24:00', 'CSGO');
      expect(target.find('.js-select').props().checked).toEqual(true);
      expectEdit(target, account);
      expect(queue.running).toHaveLength(0);
      expect(store.removeAccountMock).not.toBeCalled();
      expect(store.regenerateTokenMock).not.toBeCalled();
      expect(copyToClipboard).toHaveBeenCalledWith('7FJS3VY2273L');
      expectSnapshot(target);
    });
  });

  describe('selected and disabled regenerate', () => {
    let account: GameServerAccount;
    beforeEach(() => {
      account = new GameServerAccount({
        appid: 740,
        expired: false,
        lastLogon: '1995-12-17T03:24:00',
        memo: 'CSGO',
        steamid: '',
        token: '7FJS3VY2273L',
      });
    });

    test('initial state', () =>  {
      const target = createComponent(store, account, true, false, queue);

      expectDisabledButtons(target, false, true, false);
      expectAccountWithLastLogon(target, '7FJS3VY2273L', '740', '1995-12-17T03:24:00', 'CSGO');
      expect(target.find('.js-select').props().checked).toEqual(true);
      expectEdit(target, account);
      expect(queue.running).toHaveLength(0);
      expect(store.removeAccountMock).not.toBeCalled();
      expect(store.regenerateTokenMock).not.toBeCalled();
      expect(copyToClipboard).not.toBeCalled();
      expectSnapshot(target);
    });

    test('click regenerate', () =>  {
      const target = createComponent(store, account, true, false, queue);
      target.find('.js-regenerate').simulate('click');

      expectDisabledButtons(target, false, true, false);
      expectAccountWithLastLogon(target, '7FJS3VY2273L', '740', '1995-12-17T03:24:00', 'CSGO');
      expect(target.find('.js-select').props().checked).toEqual(true);
      expectEdit(target, account);
      expect(queue.running).toHaveLength(0);
      expect(store.removeAccountMock).not.toBeCalled();
      expect(store.regenerateTokenMock).not.toBeCalled();
      expect(copyToClipboard).not.toBeCalled();
      expectSnapshot(target);
    });
  });

  test('not selected and enabled regenerate', () =>  {
    const account = new GameServerAccount({
      appid: 740,
      expired: false,
      lastLogon: '1995-12-17T03:24:00',
      memo: 'CSGO',
      steamid: '',
      token: '7FJS3VY2273L',
    });
    const target = createComponent(store, account, false, true, queue);

    expectDisabledButtons(target, false, false, false);
    expectAccountWithLastLogon(target, '7FJS3VY2273L', '740', '1995-12-17T03:24:00', 'CSGO');
    expect(target.find('.js-select').props().checked).toEqual(false);
    expectEdit(target, account);
    expect(queue.running).toHaveLength(0);
    expect(store.removeAccountMock).not.toBeCalled();
    expect(store.regenerateTokenMock).not.toBeCalled();
    expect(copyToClipboard).not.toBeCalled();
    expectSnapshot(target);
  });

  test('not selected and disabled regenerate', () => {
    const account = new GameServerAccount({
      appid: 740,
      expired: false,
      lastLogon: undefined,
      memo: 'CSGO',
      steamid: '',
      token: '7FJS3VY2273L',
    });;
    const target = createComponent(store, account, false, false, queue);

    expect(copyToClipboard).not.toBeCalled();
    expectDisabledButtons(target, false, true, false);
    expectAccountContent(target, '7FJS3VY2273L', '740', false, 'CSGO');
    expect(target.find('.js-select').props().checked).toEqual(false);
    expectEdit(target, account);
    expect(queue.running).toHaveLength(0);
    expect(store.removeAccountMock).not.toBeCalled();
    expect(store.regenerateTokenMock).not.toBeCalled();
    expect(copyToClipboard).not.toBeCalled();
    expectSnapshot(target);
  });
});
