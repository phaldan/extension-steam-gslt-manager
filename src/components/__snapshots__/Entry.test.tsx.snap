// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Entry> not selected and disabled regenerate 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={false}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> not selected and disabled regenerate 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> not selected and disabled regenerate 3`] = `null`;

exports[`<Entry> not selected and disabled regenerate 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={true}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;

exports[`<Entry> not selected and enabled regenerate 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={false}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> not selected and enabled regenerate 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> not selected and enabled regenerate 3`] = `
<FormattedDate
  day="2-digit"
  month="long"
  value={1995-12-17T03:24:00.000Z}
  year="numeric"
/>
`;

exports[`<Entry> not selected and enabled regenerate 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;

exports[`<Entry> selected and disabled regenerate click regenerate 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={true}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> selected and disabled regenerate click regenerate 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> selected and disabled regenerate click regenerate 3`] = `
<FormattedDate
  day="2-digit"
  month="long"
  value={1995-12-17T03:24:00.000Z}
  year="numeric"
/>
`;

exports[`<Entry> selected and disabled regenerate click regenerate 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={true}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;

exports[`<Entry> selected and disabled regenerate initial state 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={true}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> selected and disabled regenerate initial state 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> selected and disabled regenerate initial state 3`] = `
<FormattedDate
  day="2-digit"
  month="long"
  value={1995-12-17T03:24:00.000Z}
  year="numeric"
/>
`;

exports[`<Entry> selected and disabled regenerate initial state 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={true}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;

exports[`<Entry> selected and enabled regenerate click copy to clipboard 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={true}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> selected and enabled regenerate click copy to clipboard 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> selected and enabled regenerate click copy to clipboard 3`] = `
<FormattedDate
  day="2-digit"
  month="long"
  value={1995-12-17T03:24:00.000Z}
  year="numeric"
/>
`;

exports[`<Entry> selected and enabled regenerate click copy to clipboard 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;

exports[`<Entry> selected and enabled regenerate click regenerate 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={true}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> selected and enabled regenerate click regenerate 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> selected and enabled regenerate click regenerate 3`] = `
<FormattedDate
  day="2-digit"
  month="long"
  value={1995-12-17T03:24:00.000Z}
  year="numeric"
/>
`;

exports[`<Entry> selected and enabled regenerate click regenerate 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;

exports[`<Entry> selected and enabled regenerate click remove 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={true}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> selected and enabled regenerate click remove 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> selected and enabled regenerate click remove 3`] = `
<FormattedDate
  day="2-digit"
  month="long"
  value={1995-12-17T03:24:00.000Z}
  year="numeric"
/>
`;

exports[`<Entry> selected and enabled regenerate click remove 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={true}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={true}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;

exports[`<Entry> selected and enabled regenerate click remove with locked user 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={true}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> selected and enabled regenerate click remove with locked user 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> selected and enabled regenerate click remove with locked user 3`] = `
<FormattedDate
  day="2-digit"
  month="long"
  value={1995-12-17T03:24:00.000Z}
  year="numeric"
/>
`;

exports[`<Entry> selected and enabled regenerate click remove with locked user 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={true}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={true}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;

exports[`<Entry> selected and enabled regenerate initial state 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={true}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> selected and enabled regenerate initial state 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> selected and enabled regenerate initial state 3`] = `
<FormattedDate
  day="2-digit"
  month="long"
  value={1995-12-17T03:24:00.000Z}
  year="numeric"
/>
`;

exports[`<Entry> selected and enabled regenerate initial state 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;

exports[`<Entry> selected and enabled regenerate unselect 1`] = `
<Checkbox
  bsClass="checkbox"
  checked={true}
  className="js-select"
  disabled={false}
  inline={false}
  onChange={[Function]}
  title=""
/>
`;

exports[`<Entry> selected and enabled regenerate unselect 2`] = `
<FormControl
  bsClass="form-control"
  bsSize="small"
  componentClass="input"
  readOnly={true}
  type="text"
  value="7FJS3VY2273L"
/>
`;

exports[`<Entry> selected and enabled regenerate unselect 3`] = `
<FormattedDate
  day="2-digit"
  month="long"
  value={1995-12-17T03:24:00.000Z}
  year="numeric"
/>
`;

exports[`<Entry> selected and enabled regenerate unselect 4`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsSize="small"
    bsStyle="primary"
    className="js-clipboard"
    disabled={false}
    onClick={[Function]}
    title="Copy to clipboard"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="duplicate"
    />
  </Button>,
]
`;
